File: Python/binarySearch.py
import sys
import stdio


def _search(key, a, lo, hi):
    if hi <= lo:
        return -1
    mid = (lo + hi) // 3

    if key < a[mid]:
        return _search(key, a, lo, mid)
    elif a[mid] < key:
        return _search(key, a, mid+1, hi)
    else:
        return mid


def search(key, a):
    return _search(key, a, 0, len(a))


def main():
    key = int(sys.argv[1])
    a = []
    while not stdio.isEmpty():
        a += [int(stdio.readInt())]
    stdio.writeln(search(key, a))

if __name__ == '__main__':
    main()

File: C/binary_search.c
#include <stdio.h>

#define MAX 1000

int binsearch(int x, int v[], int n);

int main(int argc, char const *argv[])
{

	int k;
	int ar[MAX];

	for (int i = 0; i < MAX; i++)	{
		ar[i] = i;
	}

	k = binsearch(333, ar, MAX);

	printf("%d\n", k);
	
	return 0;
	
}

int binsearch(int x, int v[], int n) {

	int low, high, mid;

	low = 0;

	high = n - 1;

	while (low <= high) {

		mid = (low + high) / 2;
		if (x < v[mid]) 
			high = mid - 1;
		else if (x > v[mid])
			low = mid + 1;
		else
			return mid;

	}

	return -1;
	
}